### PR TITLE
fixed github app link on new dashboard

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
Fixes the github app link when registering  workflows, tools, and services on the new dashboard.

The changes specifically fixes the issue where the 'Manage Dockstore Installations on GitHub' button would redirect to the homepage instead of to github. This issue was caused due to the new dashboard not having an EntryType set in its session.

**Issue**
https://ucsc-cgl.atlassian.net/browse/DOCK-2239
https://github.com/dockstore/dockstore/issues/5123

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

**Comment by svonworl** [comment]
Often, in our PRs, we include a short description of what was fixed and/or how the changes work.  If you could please add a bit of that to this PR, it would help me to better understand it.  Thanks!

File: test/web.yml [hunk 1]
```diff
@@ -25,8 +25,6 @@ orcidClientSecret: <fill me in>
 gitHubAppPrivateKeyFile: <fill me in>
 gitHubAppId: <fill me in>
 
-orcidClientID: <fill me in>
-orcidClientSecret: <fill me in>
```
Comment (kathy-t) [inline test/web.yml:29]: Why was this removed?

File: cypress/integration/group1/dashboard.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@
  *  limitations under the License.
  */
 import { resetDB, setTokenUserViewPort } from '../../support/commands';
+import { Dockstore } from '../../../src/app/shared/dockstore.model';
```
Comment (kathy-t) [inline cypress/integration/group1/dashboard.ts:17]: This is unused, can be removed

File: cypress/integration/group1/dashboard.ts [hunk 3]
```diff
@@ -43,4 +44,14 @@ describe('Dockstore dashboard', () => {
       .should('have.attr', 'href')
       .and('include', 'getting-started-with-services');
   });
+  it('Registering new tool through Github redirects correctly', () => {
```
Comment (kathy-t) [inline cypress/integration/group1/dashboard.ts:47]: Might be worth having a similar test for workflows and services too

**Review by kathy-t — COMMENTED** [review]
Thanks for fixing this for workflows and services too!

File: test/web.yml (hunk 1 above)
Comment (hyunnaye) [reply test/web.yml]: oops looks like it was removed by accident. Added it back now.

File: cypress/integration/group1/dashboard.ts (hunk 3 above)
Comment (hyunnaye) [reply cypress/integration/group1/dashboard.ts:46]: Added one for workflows. The button to register services seems to link straight to the github link so I could not add a test for it as cypress does not support different domain redirects

File: cypress/integration/group1/dashboard.ts [hunk 4]
```diff
@@ -43,4 +43,24 @@ describe('Dockstore dashboard', () => {
...
+    cy.visit('/dashboard?newDashboard');
+    cy.contains('Tools');
+    cy.get('#registerToolButton').should('be.visible').click();
+    cy.get('#3-register-workflow-option').should('be.visible').click();
+    cy.contains('button', 'Next').should('be.visible').click();
+    cy.contains('a', 'Manage Dockstore installations on GitHub')
+      .should('have.attr', 'href')
+      .and('include', 'https://github.com/apps/dockstore-testing-application');
+  });
+  it('Registering new workflow through Github redirects correctly', () => {
```
Comment (coverbeck) [inline cypress/integration/group1/dashboard.ts:56]: The blocks starting at line 47 and 57 are almost identical. You could create a function that takes a parameter(s) for what's different and avoid the duplication.

File: cypress/integration/group1/dashboard.ts [hunk 5]
```diff
@@ -43,4 +43,24 @@ describe('Dockstore dashboard', () => {
       .should('have.attr', 'href')
       .and('include', 'getting-started-with-services');
   });
+  it('Registering new tool through Github redirects correctly', () => {
+    cy.visit('/dashboard?newDashboard');
+    cy.contains('Tools');
+    cy.get('#registerToolButton').should('be.visible').click();
+    cy.get('#3-register-workflow-option').should('be.visible').click();
```
Comment (coverbeck) [inline cypress/integration/group1/dashboard.ts:50]: 1. You're copying an unfortunate existing pattern we have everywhere, but ideally, you should be using `data-cy` instead of `id` to select in Cypress. See [Cypress best practices](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements). While we don't need to go back and fix everywhere where we're using `id`, we should avoid introducing new usages of `id` as a Cypress selector, if possible.
2. The `#3` is fragile (if we change the order), although the tests will catch it. If you switch to `data-cy` you may be able to come up with a better solution.

**Comment by coverbeck** [comment]
Nice find, by the way. I know tracking down these code paths with the observables can be tricky.

File: cypress/integration/group1/dashboard.ts (hunk 5 above)
Comment (hyunnaye) [reply cypress/integration/group1/dashboard.ts]: changed from using `id` to `data-cy`

**Review by coverbeck — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]
